02-peer-protocol.md

```diff
@@ -778,6 +784,73 @@ it is simplest to only allow them to set fee levels, but as the same
 fee rate applies to HTLC transactions, the receiving node must also
 care about the reasonableness of the fee.
 
+## Message Retransmission
+
+Because communication transports are unreliable and may need to be
+re-established from time to time, the design of the transport has been
+explicitly separated from the protocol.
+
+Nonetheless, we assume that our transport is ordered and reliable;
+reconnection introduces doubt as to what has been received, so we
+retransmit any channel messages which may not have been.
+
+This is fairly straightforward in the case of channel establishment
+and close where messages have an explicit order, but in normal
+operation acknowlegements of updates are delayed until the
+`commitment_signed` / `revoke_and_ack` exchange, so we cannot assume
+the updates have been received.  This also means that the receiving
+node only needs to store updates upon receipt of `commitment_signed`.
+
+Note that messages described in [BOLT #7](07-routing-gossip.md) are
+independent of particular channels; their transmission requirements
+are covered there, and other than being transmitted after `init` (like
+any message), they are independent of requirements here.
+
+### Requirements
+
+A node MUST handle continuing a previous channel on a new encrypted
+transport.  On disconnection, a node MAY forget nodes which have not
+sent or received an `accept_channel` message.
+
+On disconnection, a node MUST reverse any uncommitted updates sent by
+the other side (ie. all messages beginning with `update_` for which no
+`commitment_signed` has been received).  A node SHOULD retain the `r`
```

Comment:
Actually, that's the other node's problem.  The real reason is that knowledge is not revocable, so as soon as the expose the payment preimage, you can (and should!) use it to claim any upstream payment.  So, really, you should have already used the `r` value...